pandas/tests/io/formats/test_format.py

```diff
@@ -3167,6 +3178,41 @@ def test_str(self):
         assert str(NaT) == "NaT"
 
 
+class TestPeriodIndexFormat:
+    def test_period_custom_locale(self, overridden_locale):
+        # GH#46319 locale-specific formatting directive leads to non-utf8 str result
+
+        # Get locale-specific reference
+        am_local, pm_local = get_local_am_pm()
+
+        # Scalar
+        per = pd.Period("2018-03-11 13:00", freq="H")
+        assert per.strftime("%p") == pm_local
+
+        # Index
+        per = pd.period_range("2003-01-01 01:00:00", periods=2, freq="12h")
+        formatted = per.format(date_format="%y %I:%M:%S%p")
+        assert formatted[0] == f"03 01:00:00{am_local}"
+        assert formatted[1] == f"03 01:00:00{pm_local}"
+
+
+@pytest.fixture(params=[None, "fr_FR", "zh_CN"])
```

Comment:
move this near the top or into the conftest.py somewhere (we might also have something like this else), pls check.

Reply:
noted, I'll have a look once the runner issue is fixed (see above)